### PR TITLE
Unit test r spec PR

### PR DESCRIPTION
**Pull request by kkasaija**
- Implemented tests for the respective classes

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require_relative '../book'
+
+describe Book do
+  context 'Book Class Unit test' do
+    it 'Create a new book when book class is instantiated' do
+      book = Book.new('Introduction to CS', 'Ruby Spec')
+      expect(book.title).to eq 'Introduction to CS'
+      expect(book.author).to eq 'Ruby Spec'
+    end
+  end
```
Comment (BenMukebo) [inline spec/book_spec.rb:12]: - [ ] Great job so far 👏🏻 👏🏻 However, according to the project requirements, there should be unit tests for all the methods in all the classes, Kindly make sure the `add_rental` classbook method has (at least) an 'it' block 👇🏻 check your book.rb file

```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: spec/classroom_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+require_relative '../classroom'
+
+describe Classroom do
+  classroom = Classroom.new('Primary 7')
+  describe 'Classroom Details' do
+    it 'should have a label of Primary 7' do
+      expect(classroom.label).to eql('Primary 7')
+    end
+  end
+end
```
Comment (BenMukebo) [inline spec/classroom_spec.rb]: - [ ] Great job so far 👏🏻 👏🏻 However, according to the project requirements, there should be an 'it' block for each class method (at least). Kindly make sure the `add_student` class classroom method has (at least) an 'it' block. check your classroom.rb file

```ruby
  def add_student(student)
    @students << student
    student.classroom = self
  end
```

File: spec/person_spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require_relative '../person'
+
+describe Person do
+  context 'Unit Test: ' do
+    it 'create a new Person when person class is instantiated' do
+      new_person = Person.new(45, 'Prof. Blessing Blast', 234)
+      expect(new_person.age).to eq 45
+      expect(new_person.name).to eq 'Prof. Blessing Blast'
+      expect(new_person.id).to eq 234
+    end
```
Comment (BenMukebo) [inline spec/person_spec.rb]: - [ ] Great job so far 👏🏻 👏🏻 However, according to the project requirements, there should be unit tests for all the methods in all the classes, Kindly make sure the `of_age?, can_use_services?,  and  correct_name` class person method has (at least) an 'it' block 👇🏻 check your person.rb file

```ruby
  def can_use_services?
    of_age? || @parent_permission
  end

  private

  def of_age?
    @age >= 18
  end

  def correct_name
    @name
  end
```

File: spec/student_spec.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+describe Student do
+  student = Student.new(30, 'Musinguzi', false)
+  describe 'Student details' do
+    it 'should be of type Student' do
+      expect(student.age).to eql(30)
+    end
+    it "should have a shape of '¯(ツ)/¯'" do
+      expect(student.play_hooky).to eql('¯(ツ)/¯')
+    end
```
Comment (BenMukebo) [inline spec/student_spec.rb:11]: - [ ] Same here, Kindly make sure the `classroom`  method has (at least) an 'it' block 👇🏻 check your student.rb file

```ruby
  def classroom=(classroom)
    @classroom = classroom
    classroom.students.push(self) unless classroom.students.include?(self)
  end
```

**Review by BenMukebo — APPROVED** [review]
### Hi @Kasaija-Kenneth, @uisendy,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻

- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- There is one file per class. ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

- [ ] Great job so far 👏🏻 👏🏻 However, according to the project requirements, there should be unit tests for all the methods in all the classes, Kindly make sure to add unit tests for all the methods of the: `Student class`, `Teacher  class`, and `Rental class`


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    it 'Person can use Service' do
+      expect(new_person.can_use_service).to eq true
+    end
+    it 'person can make rental' do
+      book = Book.new('Introduction to CS', 'Ruby Spec')
+      date = '23/01/12'
+      expect(new_person.add_rental(date, book).date).to eq('23/01/12')
+    end
```
Comment (danmainah) [inline spec/person_spec.rb]: - [ ] @team great work on your tests however you still haven't added tests for `of_age?,  and  correct_name` methods, you can use [this example](https://stackoverflow.com/questions/16599256/testing-private-method-in-ruby-rspec) on how to test private methods

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @team, 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used

 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb (hunk 5 above)
Comment (kkasaija) [reply spec/person_spec.rb]: Noted @danmainah

**Review by danmainah — APPROVED** [review]
Hi @team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kkasaija** [comment]
Much appreciated @danmainah

**Comment by danmainah** [comment]
> Much appreciated @danmainah

welcome

**Pull request merged**